lib/core/src/model.rs

```diff
@@ -724,6 +724,8 @@ pub enum ListPaymentDetails {
 pub enum GetPaymentRequest {
     /// The Lightning payment hash of the payment
     Lightning { payment_hash: String },
+    /// The SHA256-hash of the swap id in hex
+    SwapIdHash { hash: String },
```

Comment:
I think with the enum variants we wanted to keep it defined by payment methods. Maybe in this case, where the hash applies to both Lightning and Bitcoin methods, we flatten the enum into a struct and have an optional payment_hash and optional swap_id. I would keep it as swap_id and take into account in the query they could be using the id or id hash. What do you think?

Reply:
As discussed out-of-band, we can even stop requiring the distinction between identifier types. [3f5100d](https://github.com/breez/breez-sdk-liquid/pull/697/commits/3f5100dc6c796ec57353ff63d26f5c52d622f784)